qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -393,26 +393,31 @@ def _wait_for_job(self, timeout=60, wait=5):
             time.sleep(wait)
 
         if self._cancelled:
-            return Result({'id': self._id, 'status': 'CANCELLED',
-                           'result': 'job cancelled'})
+            raise JobError(
+                'Job result impossible to retrieve. The job was cancelled.')
 
         job_data = self._api.get_job(self._id)
         job_result_list = []
         for circuit_result in job_data['qasms']:
             this_result = {'data': circuit_result['data'],
                            'name': circuit_result.get('name'),
                            'compiled_circuit_qasm': circuit_result.get('qasm'),
-                           'status': circuit_result['status']}
+                           'status': circuit_result['status'],
+                           'success': circuit_result['status'] == 'DONE',
+                           'shots': job_data['shots']}
             if 'metadata' in circuit_result:
                 this_result['metadata'] = circuit_result['metadata']
 
             job_result_list.append(this_result)
 
-        return Result({'id': self._id,
-                       'status': job_data['status'],
-                       'used_credits': job_data.get('usedCredits'),
-                       'result': job_result_list,
-                       'backend_name': self.backend_name})
+        return result_from_old_style_dict({
+            'id': self._id,
+            'status': job_data['status'],
+            'used_credits': job_data.get('usedCredits'),
+            'result': job_result_list,
+            'backend_name': self.backend_name,
```

Comment:
This should be `self.backend_name()` (it was changed from a property to a method in #686).

The bug is in master now too, but this PR seems like a good place to fix.

Reply:
Good catch! I'll update to make it work with master, leaving room for re-visiting it during #800.

Reply:
Added at 4bed27d